contracts/RMRK/extension/tokenProperties/RMRKPropertiesAccessControl.sol

```diff
@@ -0,0 +1,251 @@
+     */
+    modifier onlyUnregisteredCollection(address collection) {
+        if (_issuerSettings[collection].registered) {
+            revert RMRKCollectionAlreadyRegistered();
+        }
+        _;
+    }
+
+    /**
+     * @notice Modifier to check if the caller is the issuer of the collection.
+     * @param collection Address of the collection.
+     */
+    modifier onlyIssuer(address collection) {
+        if (_issuerSettings[collection].useOwnable) {
+            if (Ownable(collection).owner() != msg.sender) {
+                revert RMRKNotCollectionIssuer();
+            }
+        } else if (_issuerSettings[collection].issuer != msg.sender) {
+            revert RMRKNotCollectionIssuer();
+        }
+        _;
+    }
+
+    /**
+     * @notice Function to check if the caller is authorized to mamage a given parameter.
+     * @dev The valid parameter types are:
+     *  [
+     *      STRING,
+     *      ADDRESS,
+     *      BYTES,
+     *      UINT,
+     *      BOOL
+     *  ]
+     * @param collection The address of the collection.
+     * @param parameterType The type of the parameter.
+     * @param parameterId The ID of the parameter.
+     * @param tokenId The ID of the token.
+     */
+    function _onlyAuthorizedCaller(
+        address collection,
+        ParameterType parameterType,
+        uint256 parameterId,
+        uint256 tokenId
+    ) private view {
+        AccessControl storage accessControl = _accessControl[collection][
+            parameterType
+        ][parameterId];
+
+        if (
+            accessControl.accessType == AccessType.Issuer &&
+            ((_issuerSettings[collection].useOwnable &&
+                Ownable(collection).owner() != msg.sender) ||
+                (!_issuerSettings[collection].useOwnable &&
+                    _issuerSettings[collection].issuer != msg.sender))
+        ) {
+            revert RMRKNotCollectionIssuer();
+        } else if (
+            accessControl.accessType == AccessType.Collaborator &&
+            !accessControl.collaboratorAddresses[msg.sender]
+        ) {
+            revert RMRKNotCollectionCollaborator();
+        } else if (
+            accessControl.accessType == AccessType.IssuerOrCollaborator &&
+            ((_issuerSettings[collection].useOwnable &&
+                Ownable(collection).owner() != msg.sender) ||
+                (!_issuerSettings[collection].useOwnable &&
+                    _issuerSettings[collection].issuer != msg.sender)) &&
+            !accessControl.collaboratorAddresses[msg.sender]
+        ) {
+            revert RMRKNotCollectionIssuerOrCollaborator();
+        } else if (
+            accessControl.accessType == AccessType.TokenOwner &&
+            IERC721(collection).ownerOf(tokenId) != msg.sender
+        ) {
+            revert RMRKNotTokenOwner();
+        } else if (
+            accessControl.accessType == AccessType.SpecificAddress &&
+            !accessControl.specificAddresses[msg.sender]
```

Comment:
Also kind of see why you're using the mapping to config the access control, but I still really dislike the bool[] + address[] interface. This might be cheaper though TBH, but more complex UX. :thinking: